### PR TITLE
[FLINK-22623][hbase] Drop BatchTableSource HBaseTableSource/Sink and related classes

### PR DESCRIPTION
**Pull request by twalthr**
## What is the purpose of the change

Removes the legacy `HBaseTableSource/Sink` which implements the BatchTableSource/Sink interface. It also removes related components i.e. `HBaseXXXInputFormat` etc.

## Brief change log

Drop legacy implementations.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): yes
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: yes
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by twalthr** [comment]
CC @JingsongLi @AHeise

File: flink-python/pyflink/table/descriptors.py [hunk 1]
```diff
@@ -33,7 +33,6 @@
     'FileSystem',
     'Kafka',
     'Elasticsearch',
-    'HBase',
```
Comment (AHeise) [inline flink-python/pyflink/table/descriptors.py:36]: Does this mean, you cannot use HBase from Python anymore? Why is it different than the other connectors?

File: flink-connectors/flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/util/HBaseTestBase.java [hunk 2]
```diff
@@ -131,7 +120,7 @@ private static void createHBaseTable1() throws IOException {
                         Timestamp.valueOf("2019-08-18 19:00:00"),
                         Date.valueOf("2019-08-18"),
                         Time.valueOf("19:00:00"),
-                        new BigDecimal(12345678.0001)));
+                        new BigDecimal("12345678.0001")));
```
Comment (AHeise) [inline flink-connectors/flink-connector-hbase-2.2/src/test/java/org/apache/flink/connector/hbase2/util/HBaseTestBase.java:123]: These changes should probably go to a different commit.

**Review by AHeise — COMMENTED** [review]
One basic question and one suggestion.

File: flink-python/pyflink/table/descriptors.py (hunk 1 above)
Comment (twalthr) [reply flink-python/pyflink/table/descriptors.py:36]: You can still use the SQL DDL for using HBase and this is also the officially documented way. The `connect()` API is outdated and requires an update (FLIP-129). I guess, initially, the Python team just adopted all "descriptors" that were present in 2019.

**Review by AHeise — APPROVED** [review]
Thanks for the clarifications. LGTM.
